src/main/java/com/vdovich/tbfm/service/impl/NasaService.java

```diff
@@ -0,0 +1,90 @@
+package com.vdovich.tbfm.service.impl;
+
+import java.io.BufferedReader;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.URL;
+
+import com.vdovich.tbfm.util.JsonObjects;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import com.vdovich.tbfm.service.INasaService;
+import com.vdovich.tbfm.util.JsonParser;
+
+@Service
+public class NasaService implements INasaService {
+
+    private static final Logger logger = LoggerFactory.getLogger(NasaService.class);
+
+    private static final String API_URL = "https://api.nasa.gov";
+    private static final String APOD = "/planetary/apod";
+    private static final String MARS_ROVER_CURIOSITY = "/mars-photos/api/v1/rovers/curiosity/latest_photos";
+    private static final String API_KEY_QUERY_PARAM = "api_key";
+    private static final String DESTINATION_FILE = "/home/user/image.jpg";
+
+
+    @Value("${nasa.api.token}")
+    private String nasaApiToken;
+
+    @Override
+    public String getPictureOfTheDay() {
```

Comment:
`getPictureOfTheDay` and `getPictureFromMars` have only one difference in code implementation - url. Please, consider avoiding duplicate code.